PMMain/Midas/MidasGUI/SCDDigester.cpp

```diff
@@ -131,7 +138,11 @@ void SCDDigester::digestKeyboardData(commandData nextCommand)
             EXACT same thing as select except that the HOLD character is used, rather than the regular character.
             // in the discusion of RingData, this would correspond to the "*Hold vectors"
             */
-
+            if (kybdGUISel % 4 == 0)
+                kybrdCtrl->setKeyChar(kybrdRingData[kybdGUISel].getRingInVector()[scdHandle->getKeySelectAngle() /  (360/kybrdRingData[kybdGUISel].getRingInVector().size())].hold);
+            else
+                kybrdCtrl->setKeyChar(kybrdRingData[kybdGUISel].getRingOutVector()[scdHandle->getKeySelectAngle() / (360/kybrdRingData[kybdGUISel].getRingOutVector().size())].hold);
```

Comment:
This logic definitely needs to be modularized. Too hard to follow in single lines.